include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -31,18 +31,67 @@ using Polygon = std::array<glm::vec3, 3>;
 
 using PolygonCache = std::shared_ptr<std::vector<Polygon>>;
 
+/// IDs of the children in a order so that we rotate the octants when moving the child with the first ID to the second
+/// ID, the second, to the third, ...
+using ChildRotationOrder = std::array<std::array<std::uint8_t, 4>, 2>;
+
+/// IDs of the edges in a order so that we rotate the cube when moving the edge with the first ID to the second ID,
+/// the second, to the third, ...
+/// As we need to update the indentations, it is required that first two indentations in the array
+/// point in the correct order towards the same corner as the second two corners. Example:
+/// Correct are either ``{{0, 1, 3, 2}}`` or ``{{3, 2, 0, 1}}``. Incorrect are all other variants.
+///
+///        -1->
+///       ^    ^
+///     0 |    | 2
+///        -3->
+using EdgeRotationOrder = std::array<std::array<std::uint8_t, 4>, 3>;
+
 class Cube : public std::enable_shared_from_this<Cube> {
     friend void ::swap(Cube &lhs, Cube &rhs) noexcept;
     template <std::size_t version>
     friend std::shared_ptr<world::Cube> io::deserialize_octree_impl(const io::ByteStream &stream);
 
 public:
-    /// Maximum of sub cubes (childs)
+    /// Maximum number of sub cubes (children, i.e. octants)
     static constexpr std::size_t SUB_CUBES = 8;
-    /// Cube edges.
+    /// Number of edges a cube has.
     static constexpr std::size_t EDGES = 12;
+
+    /// IDs of the edges in the order of for a 90° rotation around the X-axis.
+    static constexpr EdgeRotationOrder ROTATION_ORDER_EDGE_X{
+        {// First edge need to point to (crossing with) second, second to third, third to second (again) and fourth to
+         // third
+         {{2, 4, 11, 1}},
+         {{5, 7, 8, 10}},
+         {{0, 9, 6, 3}}}};
+
+    /// IDs of the edges in the order of for a 90° rotation around the Y-axis.
+    static constexpr EdgeRotationOrder ROTATION_ORDER_EDGE_Y{{{{0, 5, 9, 2}}, {{3, 8, 6, 11}}, {{1, 10, 7, 4}}}};
+
+    /// IDs of the edges in the order of for a 90° rotation around the Z-axis.
+    static constexpr EdgeRotationOrder ROTATION_ORDER_EDGE_Z{{{{1, 3, 10, 0}}, {{4, 6, 7, 9}}, {{2, 11, 8, 5}}}};
+
+    /// IDs of the children in the order of for a 90° rotation around the X-axis.
+    static constexpr ChildRotationOrder CHILD_ROTATION_ORDER_X{{{{0, 1, 3, 2}}, {{4, 5, 7, 6}}}};
+
+    /// IDs of the children in the order of for a 90° rotation around the Y-axis.
+    static constexpr ChildRotationOrder CHILD_ROTATION_ORDER_Y{{{{0, 4, 5, 1}}, {{2, 6, 7, 3}}}};
+
+    /// IDs of the children in the order of for a 90° rotation around the Z-axis.
+    static constexpr ChildRotationOrder CHILD_ROTATION_ORDER_Z{{{{0, 2, 6, 4}}, {{1, 3, 7, 5}}}};
```

Comment:
Iam still against polluting the class with those values. They are not usefull to be class member as they are only used inside the rotation function.